src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeDocblockParser.php

```diff
@@ -417,11 +420,15 @@ public static function parse(
 
         if (isset($parsed_docblock->tags['since'])) {
             $since = trim(reset($parsed_docblock->tags['since']));
-            if (preg_match('/^[4578]\.\d(\.\d+)?$/', $since)) {
-                $since_parts = explode('.', $since);
-
-                $info->since_php_major_version = (int)$since_parts[0];
-                $info->since_php_minor_version = (int)$since_parts[1];
+            // only for phpstub files or @since 8.0.0 PHP
+            // since @since is commonly used with the project version, not the PHP version
+            // https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/since.html
+            // https://github.com/vimeo/psalm/issues/10761
+            if (preg_match('/^([4578])\.(\d)(\.\d+)?(\s+PHP)?$/i', $since, $since_match)
+                && isset($since_match[1])&& isset($since_match[2])
+                && (!empty($since_match[4]) || pathinfo($code_location->file_name, PATHINFO_EXTENSION) === 'phpstub')) {
```

Comment:
It'll do for now, but we should probably limit this to only the stubs Psalm itself ships in future. When I renamed the stubs I didn't want `.phpstub` to be anything specific to Psalm, but hoped to establish a naming convention for any stubs across the ecosystem.

Reply:
Many plugins use `.phpstub` extension, and not for built-in classes / extensions: https://github.com/search?q=org%3Apsalm%20phpstub&type=code